platform/osgi/platform-osgi-configadmin/src/main/java/org/codice/felix/cm/file/EncryptingWrappedPersistenceManager.java

```diff
@@ -0,0 +1,105 @@
+import ddf.security.encryption.impl.EncryptionServiceImpl;
+
+/**
+ * Performs encryption.
+ */
+public class EncryptingWrappedPersistenceManager extends WrappedPersistenceManager {
+
+    private static final Set<String> EXCLUDED_PROPERTIES = new HashSet<>();
+
+    static {
+        EXCLUDED_PROPERTIES.add(SERVICE_PID);
+        EXCLUDED_PROPERTIES.add(SERVICE_FACTORYPID);
+        // Not worth adding a dependency on felix's fileinstall module just for the constant
+        EXCLUDED_PROPERTIES.add("felix.fileinstall.filename");
+    }
+
+    private final EncryptionService encryptionService;
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager) {
+        super(persistenceManager);
+        this.encryptionService = new EncryptionServiceImpl();
+    }
+
+    public EncryptingWrappedPersistenceManager(PersistenceManager persistenceManager,
+            EncryptionService encryptionService) {
+        super(persistenceManager);
+        this.encryptionService = encryptionService;
+    }
+
+    @Override
+    public Dictionary load(String pid) throws IOException {
+        return decryptAndReturnDictonary(super.load(pid));
+    }
+
+    @Override
+    public void store(String pid, Dictionary properties) throws IOException {
+        super.store(pid, encryptAndReturnDictonary(properties));
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> encryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> encryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            encryptedDictionaryToReturn.put(key, encryptString(key, dict.get(key)));
+        }
+        return encryptedDictionaryToReturn;
+    }
+
+    // The 2.3.7 version of the maven-bundle-plugin fails on lambdas,
+    // so we need to use Java 7 constructs
+    private Dictionary<String, Object> decryptAndReturnDictonary(Dictionary<String, Object> dict) {
+        Dictionary<String, Object> decryptedDictionaryToReturn = new Hashtable<>();
+        Enumeration<String> keys = dict.keys();
+        while (keys.hasMoreElements()) {
+            String key = keys.nextElement();
+            decryptedDictionaryToReturn.put(key, decryptString(key, dict.get(key)));
+        }
+        return decryptedDictionaryToReturn;
+    }
+
+    private Object encryptString(String key, Object value) {
+        if (value instanceof String && !EXCLUDED_PROPERTIES.contains(key)) {
+            return encryptionService.encrypt((String) value);
+        }
+        return value;
+    }
+
+    private Object decryptString(String key, Object value) {
+        if (value instanceof String && !EXCLUDED_PROPERTIES.contains(key)) {
+            return encryptionService.decrypt((String) value);
```

Comment:
❗️ The current decryption service will return the string as is if it fails to decrypt after only having logged at debug level the error. This means that we could be continuing with an encrypted value that would cause major issues elsewhere. Not sure this is acceptable.